extensions/2.0/Khronos/KHR_materials_iridescence/README.md

```diff
@@ -0,0 +1,207 @@
+
+<figure>
+<img src="./figures/ior-comparison.jpg"/>
+<figcaption><em>Comparison of different iridescence IOR values for a thin-film thickness range between 200 nm (bottom) and 800 nm (top).</em></figcaption>
+</figure>
+
+
+The iridescence effect is modeled via a microfacet BRDF with a modified Fresnel reflectance term that accounts for inter-reflections as shown in [Laurent Belcour and Pascal Barla (2017)](https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html).
+
+
+## Implementation Notes
+
+*This section is non-normative.*
+
+For the calculation of `f_specular`, the original Fresnel term `F_schlick` is exchanged with a newly introduced `F_iridescence` term. 
+To artistically control the strength of the iridescence effect, the two Fresnel terms are mixed by using the `iridescenceFactor`:
+```
+F = mix(F_schlick, F_iridescence, iridescenceFactor)
+```
+
+To calculate `F_iridescence` we need to calculate the spectral differences. 
+At first, optical path differences (`diff`) between rays reflected at the thin-film and at the base material have to be calculated:
+```
+diff = 2.0 * IOR_iridescence * thin_film_thickness * cos(viewAngle)
+```
+With the path difference, the phase shift can be calculated in Fourier space.
+
+<img src="https://render.githubusercontent.com/render/math?math=\displaystyle%20\Delta%20\phi%20=%202%20\pi%20\lambda^{-1}\text{diff}">
+
+A complete implementation for evaluating `F_iridescence` can be found in the shader file of the Khronos Sample Viewer:
+[iridescence.glsl](https://github.com/PascalSchoen/glTF-Sample-Viewer/blob/feature/KHR_materials_iridescence/source/Renderer/shaders/iridescence.glsl)
+
+### Clearcoat
+Considering KHR_materials_clearcoat extension, the clearcoat layer is evaluated before the thin-film layer. 
+Light that is passing through the clearcoat is then processed as described taking into account differences in IOR. 
+<figure>
+<img width=40% src="./figures/layering.png"/>
+<figcaption><em> </em></figcaption>
+</figure>
+
+
+## Reference
+
+### Theory, Documentation and Implementations
+
+[A Practical Extension to Microfacet Theory for the Modeling of Varying Iridescence](https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html)  
+[Arnold for Maya user guide - Thin Film](https://docs.arnoldrenderer.com/display/A5AFMUG/Thin+Film)  
+[Artist Friendly Metallic Fresnel](http://jcgt.org/published/0003/04/03/)  
+[Practical Multilayered Materials in Call of Duty: Infinite Warfare](https://blog.selfshadow.com/publications/s2017-shading-course/drobot/s2017_pbs_multilayered.pdf)  
+[Real-time Image-based Lighting of Microfacet BRDFs with Varying Iridescence](https://cg.cs.uni-bonn.de/en/publications/paper-details/kneiphof-2019-iridescence/)  
+[Real-Time Rendering, Fourth Edition; page 361ff](https://www.realtimerendering.com/)  
+[Rendering Iridescent Objects in Real-time](http://dspace.library.uu.nl/handle/1874/287110)  
```

Comment:
Would be nice to add the names of the authors and the year when the article was published. See [here](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#references) for examples.

Reply:
Thanks!